File: SimulationFront/src/app/model/Simulation.ts
import { Population } from "./Population";

export class Simulation {
  id: number;
  n: string;
  p: number;
  i: number;
  r: number;
  m: number;
  ti: number;
  tm: number;
  ts: number;

  populationList: Population[];

  constructor(
    id: number,
    n: string,
    p: number,
    i: number,
    r: number,
    m: number,
    ti: number,
    tm: number,
    ts: number,
    populationList: Population[]
  ) {
    this.id = id;
    this.n = n;
    this.p = p;
    this.i = i;
    this.r = r;
    this.m = m;
    this.ti = ti;
    this.tm = tm;
    this.ts = ts;
    this.populationList = populationList;
  }
}

File: Backend/src/main/java/com/task/Simulation/database/model/Simulation.java
package com.task.Simulation.database.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Simulation {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long id;

    String n;
    int p;
    int i;
    float r;
    float m;
    int ti;
    int tm;
    int ts;

    @OneToMany(mappedBy = "simulation", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Population> populationList;
}
